src/Service.Tests/Caching/CachingConfigDeserializationTests.cs

```diff
@@ -0,0 +1,242 @@
+            json: fullConfig,
+            out _,
+            logger: null,
+            connectionString: null,
+            replaceEnvVar: false,
+            dataSourceName: string.Empty,
+            datasourceNameToConnectionString: null,
+            replacementFailureMode: EnvironmentVariableReplacementFailureMode.Throw);
+
+        // Assert
+        Assert.IsFalse(isParsingSuccessful, message: "Expected JSON parsing to fail.");
+    }
+
+    /// <summary>
+    /// Validates that RuntimeConfigLoader.TryParseConfig(string jsonConfig) successfully deserializes
+    /// the global Runtime.Cache property.
+    /// </summary>
+    /// <param name="globalCacheConfig">Escaped JSON string defining global cache configuration.</param>
+    /// <param name="expectCacheEnabled">Whether to expect deserialized Runtime.Cache to be enabled.</param>
+    [DataRow(@"", false, DisplayName = "Global cache property left out of JSON config: default values used.")]
+    [DataRow(@",""cache"": null", false, DisplayName = "Global cache property set to Null: default values used.")]
+    [DataRow(@",""cache"": false", false, DisplayName = "Global cache property disabled -> Provided Enabled flag used with default ttl")]
+    [DataRow(@",""cache"": true", true, DisplayName = "EntityGlobal cache property enabled -> provided values used and userDefined flag set to true")]
+    [DataTestMethod]
+    public void GlobalCacheOptionsDeserialization_ValidJson(
+        string globalCacheConfig,
+        bool expectCacheEnabled)
+    {
+        // Arrange
+        string fullConfig = GetRawConfigJson(globalCacheConfig: globalCacheConfig, entityCacheConfig: string.Empty);
+
+        // Act
+        RuntimeConfigLoader.TryParseConfig(
+            json: fullConfig,
+            out RuntimeConfig? config,
+            logger: null,
+            connectionString: null,
+            replaceEnvVar: false,
+            dataSourceName: string.Empty,
+            datasourceNameToConnectionString: null,
+            replacementFailureMode: EnvironmentVariableReplacementFailureMode.Throw);
+
+        // Assert
+        Assert.IsNotNull(config, message: "Config must not be null, runtime config JSON deserialization failed.");
+        Assert.AreEqual(expected: expectCacheEnabled, actual: config.IsCachingEnabled, message: "RuntimeConfig.CacheEnabled expected to be: " + expectCacheEnabled);
+
+        if (expectCacheEnabled)
+        {
+            Assert.IsNotNull(config.Runtime?.CacheEnabled, message: "Expected global cache property to be non-null.");
+        }
+    }
+
+    /// <summary>
+    /// Validates that unexpected values provided for global runtime.cache property in the JSON config result in
+    /// a failure to deserialize the runtime config.
+    /// </summary>
+    /// <param name="globalCacheConfig">Escaped JSON string defining entity cache configuration.</param>
+    [DataRow(@",""cache"": {}", DisplayName = "Global cache property is empty: deserialization fails.")]
```

Comment:
what happens if someone puts a 1 or a 0 instead of true or false? might be a case to test here since that should be invalid.

Reply:
added these as test cases in `GlobalCacheOptionsDeserialization_InvalidJson`. 1 and 0 are not boolean values so will fail deserialization.